src/components/ArticleVoting.jsx

```diff
@@ -0,0 +1,30 @@
+import { useState } from 'react';
+import { patchArticleVotes } from '../api';
+
+export const ArticleVoting = ({ article_id, votes }) => {
+	const [voteValue, setVoteValue] = useState(0);
+	const [isError, setIsError] = useState(false);
+
+	const handleClick = (vote) => {
+		setIsError(false);
+		vote === -1
+			? setVoteValue((currVoteValue) => currVoteValue - 1)
+			: setVoteValue((currVoteValue) => currVoteValue + 1);
+
+		patchArticleVotes(article_id, vote).catch((err) => {
+			setIsError(true);
+			vote === -1
+				? setVoteValue((currVoteValue) => currVoteValue + 1)
+				: setVoteValue((currVoteValue) => currVoteValue - 1);
+		});
```

Comment:
This logic does work, but can be made simpler. The value of `vote` is what you want to use to set the value of state, and this can be done directly with this variable